pkg/morserino_com/listen_com.go

```diff
@@ -24,67 +24,126 @@ THE SOFTWARE.
-		Parity:   serial.NoParity,
-		DataBits: 8,
-		StopBits: serial.OneStopBit,
+	//If requested, use the simulator instead of a real Morserino
+	if strings.HasPrefix("SIMULATOR", strings.ToUpper(morserinoPortName)) {
+		TestMessage := "cq cq de on4kjm on4kjm = tks fer call om = ur rst 599 = hw? \n73 de on4kjm = <sk> e e"
+		Listen(iotest.OneByteReader(strings.NewReader(TestMessage)))
+		return nil
 	}
 
 	//If portname "auto" was specified, we scan for the Morserino port
 	if strings.ToUpper(morserinoPortName) == "AUTO" {
-		theComPortList, err := Get_com_list(genericEnumPorts)
+		portName, err := Get_MorserinoPort_automatically(genericEnumPorts)
 		if err != nil {
 			return err
 		}
+		morserinoPortName = portName
+	}
 
-		if theComPortList.nbrOfMorserinoPorts == 0 {
-			fmt.Println("Didn't find a connected Morserino!")
-			return fmt.Errorf("Did not find a usable port.")
-		}
-
-		if theComPortList.nbrOfMorserinoPorts > 1 {
-			fmt.Println("ERROR: Multiple Morserino devices found.")
-			return fmt.Errorf("ERROR: Multiple Morserino devices found.")
-		}
+	log.Println("Listening to port \"" + morserinoPortName + "\"")
 
-		morserinoPortName = theComPortList.morserinoPortName
-		fmt.Println("Automatically detected Morserino port: " + morserinoPortName + "\n")
+	//Port parameters for a Morserino
+	mode := &serial.Mode{
+		BaudRate: 115200,
+		Parity:   serial.NoParity,
+		DataBits: 8,
+		StopBits: serial.OneStopBit,
 	}
 
-	fmt.Println("Listening to port \"" + morserinoPortName + "\"")
-	myPort, err := genericOpenComPort.Open(morserinoPortName, mode)
+	p, err := serial.Open(morserinoPortName, mode)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
+	defer p.Close()
 
+	return Listen(p)
+}
+
+// Main receive loop
+func Listen(port io.Reader) error {
+	//TODO: needs to be moved as a goroutine
 	consoleDisplay := morserino_console.ConsoleDisplay{}
 
+	var closeKey string
+	possibleExitRequest := false
+	closeRequested := false
 	buff := make([]byte, 100)
 	for {
 		// Reads up to 100 bytes
-		n, err := myPort.Read(buff)
+		n, err := port.Read(buff)
 		if err != nil {
 			log.Fatal(err)
 		}
+
+		// Check whether the "end of transmission" was sent
+		// TODO: move this in a seperate structure/function
+		if string(buff[0:1]) == "<" {
+			closeKey = "<"
+			possibleExitRequest = true
+		} else {
+			if possibleExitRequest {
+				closeKey = closeKey + string(buff[:n])
+				if !strings.HasPrefix("<sk> e e", closeKey) {
+					possibleExitRequest = false
+				} else {
+					if closeKey == "<sk> e e" {
+						closeRequested = true
+					}
+				}
+			}
+		}
+
 		if n == 0 {
 			fmt.Println("\nEOF")
 			break
 		}
 
+		// TODO: move this out and use a channel for that
 		consoleDisplay.Add(string(buff[:n]))
+
+		if closeRequested {
+			consoleDisplay.Add("\nExiting...\n")
+			break
+		}
 	}
+
 	return nil
 }
+
+// Tries to auto detect the Morserino port
+func Get_MorserinoPort_automatically(genericEnumPorts comPortEnumerator) (string, error) {
```

Comment:
```suggestion
func DetectDevice(genericEnumPorts comPortEnumerator) (string, error) {
```

Tow nits about naming: 

- usually, golang prefers `camelCase` instead of `snake_case`, I would suggest applying this convention.
- You're already inside a packaged named `morserino`, which means that your caller would have to manipulate `morserino_com.Get_MorserinoPort_automatically`, that's a lot of morserino in here heh ?  I would suggest to always take into account your package name when declaring a package public function, here my suggestions gives `morserino_com.DiscoverDevice(...)` which is much lighter, and is still meaningful.

WDYT ?

Reply:
Very good advise ! (I am not "paid" by the number of times Morserino appears in the code....)